### ISSUE TITLE
progress callback is not working

### ISSUE THREAD

**Issue by AbdelghaniTassi** [quentinrossetti/node-7z#35]
I tried to count the number of extracted files using  the progress callback like this ;
```
.progress(function (files) {
        console.log('Some files are extracted: %s', files);
        filesNumber += files.length ;
})
```
At end the value of filesNumber is 0 even though i have 16 files in the generated directory.
I think it depends on whether 7z generates an output or not

**Comment by quentinrossetti** [comment]
Hi,
I couldn't replicate the bug. I tested for this 
```js
it('should return a countable list of files on progress', function (done) {
    var filesNumber = 0
    extract('test/zip.7z', '.tmp/test2')
    .progress(function (entries) {
      filesNumber += entries.length;
    })
    .then(function () {
      expect(filesNumber).to.be.equal(6);
      done();
    });
  });
```
and it works. 
Please give me more info about the context of your bug.
Also why makes you say 

> I think it depends on whether 7z generates an output or not
 ?

**Comment by AbdelghaniTassi** [comment]
Hi,

The test case passes because you have a 7zip version that outputs every extracted file.
When i first tested this on my development machine it did work but when i launched the integration tests on a docker container it did not work.
my counter's value did not change (always 0) and when i digged a little bit deeper i realized that the version of 7z in the docker contaner does not output any info about extracted files.

**Comment by quentinrossetti** [comment]
Ok so it appears it's the same bug as #34

**Comment by quentinrossetti** [comment]
Closed in `v1.0.0`
Check the README to adapt your code to the new API
